src/components/Toolbar.ts

```diff
@@ -0,0 +1,86 @@
+                    standardViewButtons.map(button => createElement("button", {
+                        className: "btn",
+                        onClick: () => this.props.onViewChange(button)
+                    }, button.charAt(0).toUpperCase() + button.slice(1)))
+                )
+            );
+        } else {
+            return (
+                createElement("div", { className: classNames("calendar-toolbar") },
+                    this.createGroupButton(leftButton, "left"),
+                    this.createGroupButton(centerButton, "center"),
+                    this.createGroupButton(rightButton, "right")
+                )
+            );
+        }
+    }
+
+    private createGroupButton(views: any, position: string) {
+        return createElement("div", { className: `btn-group align-${position}` },
+            views.map((view: any) => this.createToolbarElement(view))
+        );
+    }
+
+    private createToolbarElement(view: any) {
+        if (view.customView === "title") {
+            return createElement("span", { className: "calendar-label" }, this.props.label);
+        } else if (view.customView === "previous") {
+            return createElement(
+                "button", { className: "btn", title: view.buttonToolTip, onClick: () => this.props.onNavigate("PREV") },
+                createElement("span", { className: "glyphicon glyphicon-backward" })
+            );
+        } else if (view.customView === "next") {
+            return createElement(
+                "button", { className: "btn", title: view.buttonToolTip, onClick: () => this.props.onNavigate("NEXT") },
+                createElement("span", { className: "glyphicon glyphicon-forward" })
+            );
+        } else if (view.customView === "today") {
+            if (view.renderMode === "button") {
+                return createElement("button", {
+                    className: `btn btn-${view.buttonStyle}`,
+                    title: view.buttonToolTip,
+                    onClick: () => this.props.onNavigate("TODAY")
+                }, view.customCaption);
+            } else {
+                return createElement(
+                    "span", { className: "mx-link", title: view.buttonToolTip, onClick: () => this.props.onNavigate("TODAY") },
+                    createElement("a", {}, view.customCaption)
+                );
+            }
+        } else if (view.renderMode === "button") {
+            return createElement("button", {
+                className: `btn btn-${view.buttonStyle}`,
+                title: view.buttonToolTip,
+                onClick: () => this.props.onViewChange(view.customView)
+            }, view.customCaption);
+        } else {
+            return createElement(
+                "span", { className: "mx-link", title: view.buttonToolTip, onClick: () => this.props.onViewChange(view.customView) },
+                createElement("a", {}, view.customCaption)
```

Comment:
Abscract to `Button component` handeling rendering of link or button.  and use in this whole function.

Create one function for button click, you could bin the `view.customView` so inside that function you could do the if statements

Reply:
Done